modules/pico_dns_sd.c

```diff
@@ -426,10 +426,22 @@ pico_dns_sd_register_service( const char *name,
     pico_dns_sd_kv_vector_erase(txt_data);
 
     if (txt_record) {
-        pico_tree_insert(&rtree, txt_record);
+        if(pico_tree_insert(&rtree, txt_record)){
+        	if(pico_err != PICO_ERR_ENOMEM){
+        		pico_err = PICO_ERR_EINVAL;
+            }
+        	pico_tree_destroy(&rtree, NULL);
+            return -1;
+        }
```

Comment:
Don't you need to delete the `txt_record` as well? It was created a little higher, but not inserted. Also, you might want to use the `PICO_MDNS_RTREE_DESTROY` macro here, as is done later on.
(same comments apply to the code below, regarding `srv_record`)